api/src/main/java/me/lokka30/treasury/api/economy/account/Account.java

```diff
@@ -276,25 +252,26 @@ void doTransaction(
      * @param currency     the {@link Currency} of the balance being reset
      * @param subscription the {@link EconomySubscriber} accepting the new balance
      * @see PlayerAccount#resetBalance(EconomyTransactionInitiator, Currency, EconomySubscriber)
-     * @see Account#setBalance(BigDecimal, EconomyTransactionInitiator, Currency, EconomySubscriber)
      * @since v1.0.0
      */
     default void resetBalance(
             @NotNull EconomyTransactionInitiator<?> initiator,
             @NotNull Currency currency,
             @NotNull EconomySubscriber<BigDecimal> subscription
     ) {
-        setBalance(BigDecimal.ZERO, initiator, currency, new EconomySubscriber<BigDecimal>() {
-            @Override
-            public void succeed(@NotNull BigDecimal value) {
-                subscription.succeed(BigDecimal.ZERO);
-            }
+        Objects.requireNonNull(initiator, "initiator");
+        Objects.requireNonNull(currency, "currency");
+        Objects.requireNonNull(subscription, "subscription");
 
-            @Override
-            public void fail(@NotNull EconomyException exception) {
-                subscription.fail(exception);
-            }
-        });
+        doTransaction(EconomyTransaction
+                .newBuilder()
+                .withCurrency(currency)
+                .withInitiator(initiator)
+                .withTransactionAmount(BigDecimal.ZERO)
+                .withReason("reset")
+                .withImportance(EconomyTransactionImportance.NORMAL)
+                .withTransactionType(EconomyTransactionType.SET)
```

Comment:
As mentioned in the OP, we need to have these values specified by parameters in the reset method. Re-posting it here so it is not lost amongst the forthcoming reviews and comments.

Reply:
in that case we would be probably better off removing the `resetBalance` method too

Reply:
I'm lenient to go that route, however, one thing that does strike me is how the resetBalance method creates a standardized way of resetting the balance for a particular account type. For instance, PlayerAccount balances are reset to the starting balance of a currency, whilst non-PlayerAccount balances are reset to zero.

Reply:
okay lets just make the importance and reason configurable.